docs/HACKING.md

```diff
@@ -9,31 +9,62 @@ nav_order: 1
 1. TOC
 {:toc}
 
-## Building and testing in a container
 
-The majority of developers on rpm-ostree build and test it
-from a [toolbox container](https://github.com/containers/toolbox), separate from the host system.
-The instructions below may *also* work when run in a traditional
-login on a virtual machine, but are less frequently tested.
+## Setting up a development environment
 
-### Installing dependencies (cxx)
+The majority of developers on rpm-ostree build and test it from a [toolbox
+container](https://github.com/containers/toolbox), separate from the host
+system. The instructions below may *also* work when run in a traditional login
+on a virtual machine, but are less frequently tested.
 
-Today rpm-ostree uses [cxx.rs](https://cxx.rs/) - the CLI tools for
-that aren't packaged in e.g. Fedora; we ship the pre-generated
-source in the releases.  But to build from git you need to install the
-tools.
+
+### Via toolbx/Installing natively
```

Comment:
Yeah I was already wondering how to get that, thanks! I've tried to follow the new instructions in a fresh F36 container, and it ends up with this error in `make vmcheck`:

```
error: No image provided (use VMIMAGE, or cosa-builds/ or COSA_BUILDS)
```

When running `make vmcheck` in the coreos-assembler image it works. But I don't understand exactly what is different there... ?

Reply:
Yeah, this is confusing. To summarize:
- in the toolbx/self-install path, you'd install cosa inside your container, which will give you `cosa` and `kola`
- in the buildroot container path, you'd build rpm-ostree using the buildroot container, but run the kola tests using [the canonical cosa container](https://quay.io/repository/coreos-assembler/coreos-assembler?tab=tags). For the `vmcheck` tests, you should be able to run `tests/vmcheck.sh` directly from the cosa container. We're slowly working on migrating our vmcheck tests to become kola tests to make this less awkward.

Reply:
It's really tempting though to have the buildroot container just build and install cosa as well. We can't directly derive `FROM` it because the whole purpose of moving it to fedora-coreos-config was to decouple it (see https://github.com/coreos/fedora-coreos-config/pull/740). But cosa should be flexible enough to support being installed in at least Fedora N and Fedora N+1 userspaces (we really should add CI for that).

In fact, that's what I do myself in my pet container which I build for stable Fedora up to rawhide: https://github.com/jlebon/pet/blob/8c012dba5ccbaa4c9305c2f9effafefc3536e20b/build#L7-L22.